### ISSUE TITLE
Request: Mark concretely-typed fields as unusual

### ISSUE THREAD

**Issue by ParadaCarleton** [julia-vscode/StaticLint.jl#366]
e.g. Float64 here:
![image](https://github.com/julia-vscode/StaticLint.jl/assets/71727937/c5322d87-e4ba-4d2e-849b-bfcb336a2479)

**Comment by fredrikekre** [comment]
Why?

**Comment by ParadaCarleton** [comment]
> Why?

Sorry, made a mistake in the title--this is for function arguments, although it might also apply (to a slightly lesser extent) to fields in a struct.

The reason I'd like a blue squiggly is because new users often make the mistake of overtyping their functions. It's very rare that you actually need a function to _only_ allow Float64s.

**Comment by dzhang314** [comment]
I strongly disagree with this suggestion. I write lots of high-performance code (e.g., [**MultiFloats.jl**](https://github.com/dzhang314/MultiFloats.jl)) that depends on exact bitwise representation, so my functions are correctly and appropriately specialized to concrete types.

Maybe _you_ don't write code like that, and that's fine, but I think this is _way_ too opinionated to turn on in the linter by default.
